flask_webgoat/auth.py

```
@@ -14,12 +10,13 @@ def login():
 
-    # vulnerability: SQL Injection
-    query = (
-        "SELECT id, username, access_level FROM user WHERE username = '%s' AND password = '%s'"
-        % (username, password)
-    )
-    result = query_db(query, [], True)
+    # Use parameterized query to avoid SQL injection
+    query = "SELECT id, username, access_level FROM user WHERE username = ? AND password = ?"
+    try:
+        result = query_db(query, (username, password), True)  # PRECOGS_FIX: use parameterized query instead of string interpolation to prevent SQL injection
+    except Exception:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)